src/libs/Navigation/switchPolicyID.ts

```diff
@@ -93,51 +94,31 @@ export default function switchPolicyID(navigation: NavigationContainerRef<RootSt
         return;
     }
 
+    // The correct route for SearchPage is located in the CentralPane
+    const shouldAddToCentralPane = !getIsNarrowLayout() || isOpeningSearchFromBottomTab;
+
     // If the layout is wide we need to push matching central pane route to the stack.
-    if (!getIsNarrowLayout()) {
-        // Case when the user selects "All" workspace from the specific workspace settings
-        if (checkIfActionPayloadNameIsEqual(actionForBottomTabNavigator, SCREENS.ALL_SETTINGS) && !policyID) {
-            root.dispatch({
-                type: CONST.NAVIGATION.ACTION_TYPE.PUSH,
-                payload: {
-                    name: NAVIGATORS.CENTRAL_PANE_NAVIGATOR,
-                    params: {
-                        screen: SCREENS.SETTINGS.WORKSPACES,
-                        params: undefined,
-                    },
-                },
-            });
-        } else {
-            const topmostCentralPaneRoute = getTopmostCentralPaneRoute(rootState);
-            const screen = topmostCentralPaneRoute?.name;
-            const params: CentralPaneRouteParams = {...topmostCentralPaneRoute?.params};
-            const isWorkspaceScreen = screen && Object.values(SCREENS.WORKSPACE).includes(screen as ValueOf<typeof SCREENS.WORKSPACE>);
-
-            // Only workspace settings screens have to store the policyID in the params.
-            // In other case, the policyID is read from the BottomTab params.
-            if (!isWorkspaceScreen) {
-                delete params.policyID;
-            } else {
-                params.policyID = policyID;
-            }
-
-            // If the user is on the home page and changes the current workspace, then should be displayed a report from the selected workspace.
-            // To achieve that, it's necessary to navigate without the reportID param.
-            if (checkIfActionPayloadNameIsEqual(actionForBottomTabNavigator, SCREENS.HOME)) {
-                delete params.reportID;
-            }
-
-            root.dispatch({
-                type: CONST.NAVIGATION.ACTION_TYPE.PUSH,
-                payload: {
-                    name: NAVIGATORS.CENTRAL_PANE_NAVIGATOR,
-                    params: {
-                        screen,
-                        params,
-                    },
-                },
-            });
```

Comment:
This code is related to the pages that have been already refactored. Previously Workspace Switcher was available on Account Settings and Workspace Initial Page pages, we don't need this anymore since these pages don't display WS :)